pandas_gbq/gbq.py

```diff
@@ -188,78 +172,103 @@ def get_application_default_credentials(self):
+            credentials do not have access to the project (self.project_id)
+            on BigQuery.
+        """
+        with open('bigquery_credentials.dat') as credentials_file:
+            credentials_json = json.load(credentials_file)
+            credentials = google.oauth2.credentials.Credentials(
+                token=credentials_json.get('access_token'),
+                refresh_token=credentials_json.get('refresh_token'),
+                id_token=credentials_json.get('id_token'),
+                token_uri=credentials_json.get('token_uri'),
+                client_id=credentials_json.get('client_id'),
+                client_secret=credentials_json.get('client_secret'),
+                scopes=credentials_json.get('scopes'))
+
+            # Refresh the token before trying to use it.
+            http = httplib2.Http()
+            request = google_auth_httplib2.Request(http)
+            credentials.refresh(request)
+
+        return self.try_credentials(credentials)
+
+    def save_user_account_credentials(self, credentials):
+        """
+        Saves user account credentials to a local file.
+        """
+        with open('bigquery_credentials.dat', 'wb') as credentials_file:
+            credentials_json = {
+                'refresh_token': credentials.refresh_token,
+                'id_token': credentials.id_token,
+                'token_uri': credentials.token_uri,
+                'client_id': credentials.client_id,
+                'client_secret': credentials.client_secret,
+                'scopes': credentials.scopes,
+            }
+            json.dump(credentials_file, credentials_json)
 
-        flow = OAuth2WebServerFlow(
-            client_id=('495642085510-k0tmvj2m941jhre2nbqka17vqpjfddtd'
-                       '.apps.googleusercontent.com'),
-            client_secret='kOc9wMptUtxkcIFbtZCcrEAc',
-            scope=self.scope,
-            redirect_uri='urn:ietf:wg:oauth:2.0:oob')
+    def get_user_account_credentials(self, auth_local_webserver=True):
+        """Gets user account credentials.
 
-        storage = Storage('bigquery_credentials.dat')
-        credentials = storage.get()
+        This method authenticates using user credentials, either loading saved
+        credentials from a file or by going through the OAuth flow.
 
-        if credentials is None or credentials.invalid or self.reauth:
-            credentials = run_flow(flow, storage, argparser.parse_args())
+        Parameters
+        ----------
+        auth_local_webserver : boolean (default True)
+            Use a local webserver to complete the OAuth flow.
 
-        return credentials
+        Returns
+        -------
+        GoogleCredentials : credentials
+            Credentials for the user with BigQuery access.
+        """
+        credentials = self.load_user_account_credentials()
+
+        client_config = {
+            'installed': {
+                'client_id': ('495642085510-k0tmvj2m941jhre2nbqka17vqpjfddtd'
```

Comment:
Because it was hard coded previously. I just converted the variable to the expected format. This config variable is how Google knows to say Pandas GBQ is requesting access versus some other project.